### PR TITLE
Add CARMA-JAX as an in-process box model (coagulation)

### PR DESCRIPTION
**Pull request by aliakherati**
## Summary

Adds [CARMA-JAX](https://github.com/reflective-org/carma-jax) as a supported box model: a JAX port of the CARMA sectional aerosol scheme, resolving the size distribution on a configurable geometric mass grid (default: 47 mass-doubling bins, radii ~0.2 nm – 8 µm).

Like the other JAX models it runs **in process**: `create_input` → `run`/`run_ensemble` → the standard `analysis.Output`, plus a module-level `retrieve_model_state` mirroring the PartMC/MAM4 functions. Three commits; details in [`CARMA_JAX.md`](CARMA_JAX.md), runnable walkthrough in [`demo_carma_jax.ipynb`](demo_carma_jax.ipynb) (no binaries needed; outputs stripped).

## Scope — deliberate and explicit

This wires CARMA-JAX's **coagulation**, its production-quality path: a single internally-mixed involatile group, stepped by a fully JIT-compiled kernel, with the physical Brownian coagulation kernel built from the scenario's temperature and pressure exactly as CARMA's own drivers build it (`setup_atm → setup_vf → setup_ckern`).

CARMA-JAX's condensational growth and sulfate nucleation exist on its `dev` branch, but the environment builder they require lives in that repository's `scripts/` and is not installed with the package. Rather than vendor unowned physics code or silently no-op, the adapter **raises `ValueError`** when those processes are requested. Once CARMA-JAX exports its environment builder, wiring them is a contained follow-up.

The dependency is pinned to a `dev` commit: `dev` is the active branch, and the curated `main` release is coagulation-only with older, pre-Fortran-parity bin-edge formulas.

## Design notes

- **No changes to shared framework code** — one new module plus one line each in `ambrs/__init__.py` and `requirements.txt`.
- **Binning is exact:** each log-normal mode is placed by the analytic integral between every bin's boundary radii, so total number is preserved by construction (a warning fires if a mode falls off the grid).
- **Composition:** CARMA carries one internally-mixed element, so multi-mode compositions are blended volume-weighted into one mixture (warned when the modes actually differ); the blend is reported uniformly across the output population.
- **Bin grid is fixed at model construction** (`nbin`, `rmin`, mass ratio, density) because CARMA's coagulation pair tables are built from it — a ~1 s one-time cost, after which the compiled step is reused across the whole ensemble.
- Dry mass is conserved to machine precision under coagulation (~1e-15 over a 24 h run); the tests pin this.

## Testing

19 new tests in `test/test_carma_jax.py`, same `unittest` style as the suite:

- **With carma-jax installed:** `95 passed` (76 baseline + 19 new).
- **Without it:** `76 passed, 19 skipped` — `import ambrs` still works (guard catches `ImportError` and `OSError`).

## Demo

`demo_carma_jax.ipynb` shows a day of pure coagulation on CARMA's own bin grid, the invariants (number decays; mass flat to 1.5e-15), and a bin-resolution comparison at fixed size span — including the one-moment lesson that coarse grids misstate total mass from step 0, since particles carry the bin-centre mass. That resolution sensitivity is exactly the kind of structural difference AMBRS exists to expose.

## Dependency note

carma-jax is public and installs over `git+https`; it pulls in `jax`/`jaxlib`. If you'd rather keep those optional, I'll move it to an optional requirements file — the import guard already makes the tests skip cleanly.

**Overlap note:** this PR, #99 and #100 each add one line to `ambrs/__init__.py`, `requirements.txt` and the README's model list; whichever merge later need a trivial rebase.

Suggested labels: `box model integration`, `enhancement`, `python`, `dependencies`, `testing`.

### CODE REVIEW

**Comment by GlazerMann** [comment]
If I'm reading the code correctly, it applies volume-weighting to mass fractions using modal volume. But if the species/modes have different effective densities, that's not generally equivalent to mass-weighting.

**Comment by GlazerMann** [comment]
One output-state behavior seems worth clarifying.

This CARMA-JAX configuration deliberately has no gas phase:

```python
ngas = 0
```

and therefore does not evolve the Scenario's gas concentrations. However, `run()` currently returns:

```python
gas_mixture = build_gas_mixture({'units': 'kg_per_kg'})
```

which produces a zero gas mixture regardless of the gases present in the input Scenario.

For a coagulation-only model, would it be more consistent for the AMBRS `Output` to preserve the unchanged input gas mixture? Otherwise a Scenario containing nonzero SO2 or H2SO4 appears to lose those gases even though CARMA never processed them.

A regression using nonzero input gas concentrations would make the intended contract explicit: either verify that unaffected gases are carried through, or document that CARMA-JAX outputs intentionally report no gas state.

I would not treat this as necessarily a CARMA physics bug; it is mainly a question of what AMBRS `Output` is supposed to represent for state variables that the selected model does not evolve.
